avalon/tools/loader/app.py

```diff
@@ -249,6 +253,9 @@ def on_refreshed(has_item):
 
         self.data["state"]["assetIds"] = asset_ids
 
+        representations = self.data["widgets"]["representations"]
+        representations.model.set_version_ids([])  # reset repre list
```

Comment:
It is better not call methods on attribute of widget. Rather implement `set_version_ids` on representations widget that wraps that.
```
def set_version_ids(self, version_ids):
    self.model.set_version_ids(version_ids)
````

This way you'll prevent touching of inner logic of representations widget in future. Like when you'll need to do something before it is called on model.